packages/design-tokens/scripts/style-dictionary.js

```diff
@@ -0,0 +1,495 @@
+
+/**
+ * transform: scss variable names
+ * example: `$namespace-item-variant-property-modifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/scss',
+  type: 'name',
+  transformer: pathToKebabCase,
+})
+
+/**
+ * transform: css variable names
+ * example: `--namespace-item-variant-property-modifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/css',
+  type: 'name',
+  transformer: pathToKebabCase,
+})
+
+/**
+ * transform: js variable names
+ * example: `namespace.item.variant.property.modifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/js',
+  type: 'name',
+  transformer: pathToDotNotation,
+})
+
+/**
+ * transform: js es6 variable names
+ * example: `NamespaceItemVariantPropertyModifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/js/es6',
+  type: 'name',
+  transformer: pathToPascalCase,
+})
+
+// find values with px unit
+function isPx(value) {
+  return /[\d.]+px$/.test(value)
+}
+
+// transform: px to rem
+StyleDictionary.registerTransform({
+  name: 'pxToRem',
+  type: 'value',
+  transformer: token => {
+    if (isPx(token.value)) {
+      const baseFontSize = 16
+      const floatValue = parseFloat(token.value.replace('px', ''))
+
+      if (isNaN(floatValue)) {
+        return token.value
+      }
+
+      if (floatValue === 0) {
+        return '0'
+      }
+
+      if (token.name.includes('lineHeight')) {
+        return `${floatValue / baseFontSize}`
+      }
+
+      return `${floatValue / baseFontSize}rem`
+    }
+    return token.value
+  },
+})
+
+//-----
+
+// ts output
+StyleDictionary.registerTransform({
+  name: 'attribute/typescript',
+  type: 'attribute',
+  transformer: token => {
+    return {
+      typescript: {
+        // these transforms will need to match the ones you use for typescript
+        // or you can "chain" the transforms and use token.name and token.value
+        // like scss and less transforms do.
+        name: token.path.slice(1).join('.'),
+        value: token.value,
+      },
+    }
+  },
+})
+
+// css output
+StyleDictionary.registerTransform({
+  name: 'attribute/css',
+  type: 'attribute',
+  transformer: token => {
+    const tokenName = token.path.slice(1).join('-')
+    return {
+      css: {
+        name: `--${tokenName}`,
+        value: token.value,
+      },
+    }
+  },
+})
+
+// transform: composite typography to shorthands
+StyleDictionary.registerTransform({
+  name: 'typography/shorthand',
+  type: 'value',
+  transitive: true,
+  matcher: token => token.type === 'typography',
+  transformer: token => {
+    const {value} = token
+
+    // if lineHeight has value, include in shorthand
+    if (value.lineHeight) {
+      return `${value.fontWeight} ${value.fontSize}/${value.lineHeight} ${value.fontFamily}`
+    }
+
+    return `${value.fontWeight} ${value.fontSize} ${value.fontFamily}`
+  },
+})
+
+// REGISTER THE CUSTOM TRANFORM GROUPS
+
+StyleDictionary.registerTransformGroup({
+  name: 'css',
```

Comment:
Just FYI, you don't need to use TransformGroups, you can just add the transforms to the platform directly.